src/components/network-chart/index.js

```diff
@@ -61,80 +60,104 @@ class NetworkChart extends React.PureComponent<Props> {
     return interval / (end - start);
   }
 
-  render() {
-    const {
-      maxNetworkRows,
-      timeRange,
-      threadIndex,
-      networkTimingRows,
-      markers,
-      previewSelection,
-      updatePreviewSelection,
-    } = this.props;
+  _getMarkerPosition = markerStart => {
+    const timeRange = this.props.timeRange;
+    const timeRangeTotal = timeRange.end - timeRange.start;
+
+    let markerPosition =
+      (markerStart - timeRange.start) *
+      (100 - CONTAINER_WIDTH) /
+      timeRangeTotal;
+
+    // the bar shall not overlap the first 150px as this is the thread label area.
+    if (markerPosition < 0) {
+      markerPosition = 0;
+    }
+
+    markerPosition = markerPosition + CONTAINER_WIDTH;
+
+    return markerPosition;
+  };
 
-    // The viewport needs to know about the height of what it's drawing, calculate
-    // that here at the top level component.
-    const maxViewportHeight = maxNetworkRows * ROW_HEIGHT;
+  _getMarkerLength = markerDuration => {
+    const timeRange = this.props.timeRange;
+    const timeRangeTotal = timeRange.end - timeRange.start;
+
+    let markerLength =
+      markerDuration * (100 - CONTAINER_WIDTH) / timeRangeTotal;
```

Comment:
Checking for the relation of `marker.dur`and `timeRangeTotal` to get the length of the `itemBar` in percent.

We subtract the `CONTAINER_WIDTH` to align with the timethread above.